flocker/control/httpapi.py

```diff
@@ -796,6 +799,38 @@ def delete_container_configuration(self, name):
         # Didn't find the application:
         raise CONTAINER_NOT_FOUND
 
+    @app.route("/configuration/_compose", methods=['POST'])
+    # Undocumented for now since this is temporary, private endpoint for
+    # flocker-deploy.
+    @structured(
+        inputSchema={
+            '$ref':
+            '/v1/endpoints.json#/definitions/configuration_compose'
+        },
+        outputSchema={},
+        schema_store=SCHEMAS
+    )
+    def replace_configuration(self, applications, deployment):
+        """
+        Replace the existing configuration with one given by flocker-deploy
+        command line tool.
+
+        :param applications: Configuration in Flocker-native or
+            Fig/Compose format.
+
+        :param deployment: Configuration of which applications run on
+            which nodes.
+        """
+        try:
+            configuration = FigConfiguration(applications)
+            if not configuration.is_valid_format():
+                configuration = FlockerConfiguration(applications)
+            return self.persistence_service.save(model_from_configuration(
```

Comment:
Our conversation elsewhere concluded that it's a change in quantity rather than quality.  `flocker-deploy` may already complete before your application is running.  This change might make it a little more likely (depending what configuration you're deploying).  This is neither entirely good or bad.  Coupled with a change to `flocker-deploy` to tell the user that they might have to wait a while, it's probably a net positive.  So we'll just go with that.